plasmoid/contents/ui/ConfigGeneral.qml

```diff
@@ -17,22 +17,76 @@ Item {
     width: childrenRect.width
     height: childrenRect.height
 
+    property alias cfg_locationEntry: locationEntry.checked
+    property alias cfg_woeidEntry: woeidEntry.checked
     property alias cfg_woeid: woeidField.text
     property alias cfg_interval: intervalField.text
     property alias cfg_timeFormat24: timeFormat24Field.checked
     property alias cfg_useWxFonts: useWxFontsField.checked
+    property bool locChecked: plasmoid.configuration.locationEntry
+    property bool woeChecked: plasmoid.configuration.woeidEntry
+
+    function onLocClicked() {
+        if (woeChecked && plasmoid.configuration.locationEntry) { 
+            cfg_woeid = plasmoid.configuration.woeid
```

Comment:
You don't assign value again here. `cfg_` is a special prefix for Plasma and the configuration value will be set automatically (since we have already declared it as a `alias`).